pkg/helpers/config/userdata.yaml

```diff
@@ -1,17 +1,17 @@
 #cloud-config
-repo_update: true
 write_files:
   - path: /run-container.sh
     permissions: 755
     content: |
       #!/bin/bash
-      # based on tls, set up docker run command.
       echo "${USERDATA_BEGIN}" >> /var/log/userdata-output
-      sudo docker pull ${VALIDATOR_IMAGE} || echo "Warning: could not pull the specified docker image, will try to use the prepulled one" >> /var/log/userdata-output
-      VALIDATOR_REFERENCE=`echo ${VALIDATOR_IMAGE} | cut -d : -f 1`
-      # Retrieving the latest image successfully pulled (either from the script, or prepulled in the AMI)
+      # Look for the image pre-pulled in the AMI, if not available, try to pull it
       IMAGE=`docker images ${VALIDATOR_REPO} -q  | head -n 2 | tail -n 1`
```

Comment:
should be VALIDATOR_REFERENCE (not VALIDATOR_REPO, this is a typo) - https://issues.redhat.com/browse/OSD-13343 
I started doing some testing on this, but as you are updating this part, would make sense to update it all together.